libraries/AP_HAL_Linux/AnalogIn_Imx_IOMCU.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "AP_HAL_Linux.h"
+#include "Imx_IOMCU.hpp"
+
+extern const AP_HAL::HAL &hal;
+
+class AnalogSource_Imx_IOMCU : public AP_HAL::AnalogSource {
+public:
+	friend class AnalogIn_Imx_IOMCU;
+	AnalogSource_Imx_IOMCU(int16_t pin)
+	{
+		_pin = pin;
+		_value = 0;
+	}
+	float read_average()
+	{
+		return read_latest();
+	}
+	float read_latest()
+	{
+		return _value;
+	}
+	void set_pin(uint8_t p)
+	{
+		_pin = p;
+	}
+	void set_stop_pin(uint8_t p) {}
+	void set_settle_time(uint16_t settle_time_ms) {}
+	float voltage_average()
+	{
+		return _value;
+	}
+	float voltage_latest()
+	{
+		return _value;
+	}
+	float voltage_average_ratiometric()
+	{
+		return _value;
+	}
+private:
+	int16_t _pin;
+	float _value;
+};
+
+class AnalogIn_Imx_IOMCU : public AP_HAL::AnalogIn {
+public:
+	AnalogIn_Imx_IOMCU(Imx_IOMCU *m)
+	{
+		mcu = m;
+	}
+
+	void init() override
+	{
+		hal.scheduler->register_timer_process(FUNCTOR_BIND_MEMBER(&AnalogIn_Imx_IOMCU::_update, void));
```

Comment:
unless it's a simple assignment or trivial, please move the implementation to a `.cpp` file with the same name

Reply:
ok,got it.